src/github/pullRequestManager.ts

```diff
@@ -647,31 +639,66 @@ export class PullRequestManager implements vscode.Disposable {
 			};
 		}
 
-		if (!options.fetchNextPage) {
-			for (const repository of this._githubRepositories) {
-				this._repositoryPageInformation.set(repository.remote.url.toString(), {
-					pullRequestPage: 1,
+		const queryId = type.toString() + (query || '');
+		for (const repository of this._githubRepositories) {
+			const remoteId = repository.remote.url.toString() + queryId;
+			if (!this._repositoryPageInformation.get(remoteId)) {
+				this._repositoryPageInformation.set(remoteId, {
+					pullRequestPage: 0,
 					hasMorePages: null
 				});
 			}
 		}
 
+		const pullRequestData: PullRequestData = { hasMorePages: false, pullRequests: [] };
+		const addPage = (page: PullRequestData | undefined) => {
+			if (page) {
+				pullRequestData.pullRequests = pullRequestData.pullRequests.concat(page.pullRequests);
+				pullRequestData.hasMorePages = page.hasMorePages;
+			}
+		};
+
 		const githubRepositories = this._githubRepositories.filter(repo => {
-			const info = this._repositoryPageInformation.get(repo.remote.url.toString());
-			return info && info.hasMorePages !== false;
+			const info = this._repositoryPageInformation.get(repo.remote.url.toString() + queryId);
+			// If we are reloading (fetchNextPage === false), don't filter out repos that are out of pages.
+			return info && (options.fetchNextPage === false || info.hasMorePages !== false);
 		});
 
 		for (let i = 0; i < githubRepositories.length; i++) {
 			const githubRepository = githubRepositories[i];
-			const pageInformation = this._repositoryPageInformation.get(githubRepository.remote.url.toString())!;
-			const pullRequestData = type === PRType.All
-				? await githubRepository.getAllPullRequests(pageInformation.pullRequestPage)
-				: await githubRepository.getPullRequestsForCategory(query || '', pageInformation.pullRequestPage);
+			const remoteId = githubRepository.remote.url.toString() + queryId;
+			const pageInformation = this._repositoryPageInformation.get(remoteId)!;
+
+			const fetchPage = async (pageNumber: number) =>
+				type === PRType.All
+					? await githubRepository.getAllPullRequests(pageNumber)
+					: await githubRepository.getPullRequestsForCategory(query || '', pageNumber);
+
+			if (options.fetchNextPage) {
+				pageInformation.pullRequestPage++;
+				addPage(await fetchPage(pageInformation.pullRequestPage));
+			} else {
+				if (pageInformation.pullRequestPage === 0) { pageInformation.pullRequestPage = 1; }

```

Comment:
should `pullRequestPage` be initialized to 1 above instead?

Reply:
That would cause the "continue searching in other remotes" action, which triggers a "fetchNextPage" command, to pull from page 2. 

This whole thing is a bit tricky because there are three behaviours:
1) Initialize: fetch the first page of the first remote that has pages
2) Fetch Next: fetch the next page from this remote, or if it has no more pages, the first page from the next remote that does have pages
3) Restore (new): fetch all the pages you previously have fetched

And only 2 options: `fetchNextPage: true | false`.

So when `fetchNextPage: true` we do 2, and when it's `false`, we do either 1 or 3. The initial state of 0 `pullRequestPage`s fetched so far is what distinguishes 1 and 3.

Working on a better solution now...

Reply:
Just pushed a reworking of the logic.